op-node/node/node.go

```diff
@@ -483,30 +487,35 @@ func (n *OpNode) initPProf(cfg *Config) error {
 	return nil
 }
 
-func (n *OpNode) initP2P(ctx context.Context, cfg *Config) error {
-	if cfg.P2P != nil {
+func (n *OpNode) p2pEnabled() bool {
+	return n.cfg.P2P != nil
```

Comment:
Don't you also need to check that it's actually enabled?
```suggestion
	return n.cfg.P2P != nil && !n.cfg.P2P.Disabled()
```